### PR TITLE
Revert "Move code coverage steps into shared template. Enable live test opt-in"

### PR DESCRIPTION
**Pull request by benbp**
Reverts Azure/azure-sdk-for-net#21277

I accidentally merged this PR, forgetting that were some outstanding coverage issues I was seeing and needed to debug (but not captured by build failures).

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 1]
```diff
@@ -43,15 +43,12 @@ jobs:
       - template: /eng/common/pipelines/templates/steps/verify-agent-os.yml
         parameters:
           AgentImage: $(OSVmImage)
-
```
Comment (heaths) [inline eng/pipelines/templates/jobs/ci.tests.yml:46]: Nit: IMO, I do like the extra line break for clarity, but not a big deal.

**Review by heaths — APPROVED** [review]
Small nit, but I suppose this is just a revert. When you do the eventual merge, consider the nit. :)

File: eng/pipelines/templates/jobs/ci.tests.yml (hunk 1 above)
Comment (benbp) [reply eng/pipelines/templates/jobs/ci.tests.yml:46]: When I revert the revert I'll keep the breaks in.

**Pull request merged**